libcfnet/client_protocol.c

```diff
@@ -196,7 +199,10 @@ static bool SetSessionKey(AgentConnection *conn)
         return false;
     }
 
-    conn->session_key = (unsigned char *) bp->d;
+    conn->session_key = xmalloc(BN_num_bytes(bp));
+    BN_bn2bin(bp, conn->session_key);
```

Comment:
So previously `session_key` was the internal representation directly, but now we convert it using `BN_bn2bin()` which [is not just a plain byte copy (or is it)](https://github.com/openssl/openssl/blob/OpenSSL_1_1_0f/crypto/bn/bn_lib.c#L521). So this could be a behaviour change. It needs to be checked with classic protocol connections between old CFEngine versions and new ones.